src/transformers/models/wav2vec2_with_lm/processing_wav2vec2_with_lm.py

```diff
@@ -93,6 +93,11 @@ def __init__(
         if not isinstance(decoder, BeamSearchDecoderCTC):
             raise ValueError(f"`decoder` has to be of type {BeamSearchDecoderCTC.__class__}, but is {type(decoder)}")
 
+        if feature_extractor.__class__.__name__ not in ["Wav2Vec2FeatureExtractor", "SeamlessM4TFeatureExtractor"]:
+            raise ValueError(
+                f"`feature_extractor` has to be of type `Wav2Vec2FeatureExtractor` or `SeamlessM4TFeatureExtractor`, but is {type(feature_extractor)}"
+            )
+
```

Comment:
Not sure if this is the most transformers-like way of checking this, WDYT ?

Reply:
LGTM!

Reply:
Perhaps it would be more general if we add this to the `ProcessorMixin` class rather than manually defining this ValueError (as we have this case for various models). Could be called supported_feature_extractors for instance, and the mixin class would then automatically raise a `ValueError` in case the class is not supported

Reply:
I think Auto classes are only used for two processors at the moment (neither merged)? If we enable using autoclasses and then add authentication into the ProcessorMixin, then we're just doing a more round-about alternative to just being able to specify more than one class for `feature_extractor_class`, which I think would be preferable.